staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -230,14 +226,14 @@ func (a *auditResponseWriter) Unwrap() http.ResponseWriter {
 
 func (a *auditResponseWriter) processCode(code int) {
 	a.once.Do(func() {
-		if a.event.ResponseStatus == nil {
-			a.event.ResponseStatus = &metav1.Status{}
+		ac := audit.AuditContextFrom(a.ctx)
+		if ac.GetEventResponseStatus() == nil {
+			ac.SetEventResponseStatus(&metav1.Status{})
 		}
-		a.event.ResponseStatus.Code = int32(code)
-		a.event.Stage = auditinternal.StageResponseStarted
+		ac.GetEventResponseStatus().Code = int32(code)
```

Comment:
nit: avoid the extra mutex operation

```go
if ac.GetEventResponseStatus() == nil {
	ac.SetEventResponseStatus(&metav1.Status{Code: int32(code)})
} else {
	ac.GetEventResponseStatus().Code = int32(code)
}
```

Reply:
going with the following, hope that's ok :) i hate seeing the `ac.GetEventResponseStatus().Code` pattern

```
if status := ac.GetEventResponseStatus(); status == nil {
	ac.SetEventResponseStatus(&metav1.Status{Code: int32(code)})
} else {
	status.Code = int32(code)
}
```

Reply:
Hmm, status is technically unguarded here. Maybe we just need a `SetEventResponseStatusCode` method? blech.

Reply:
Ugh indeed. but added it. thanks!